contracts/BullaFinance.sol

```diff
@@ -0,0 +1,131 @@
+
+    /// @param _fee the new fee
+    /// @notice SPEC:
+    ///     allows an admin to update the fee amount to `_fee`
+    ///     Given the following: `msg.sender == admin`
+    function feeChanged(uint256 _fee) public virtual;
+
+    /// @notice SPEC:
+    ///     allows an admin to withdraw `withdrawableFee` amount of tokens from this contract's balance
+    ///     Given the following: `msg.sender == admin`
+    function withdrawFee() public virtual;
+
+    //
+    //// CREDITOR FUNCTIONS ////
+    //
+
+    /// @param claim claim creation parameters
+    /// @param terms financing terms
+    /// @notice SPEC:
+    ///     Allows a user to create a Bulla Claim with and offer finance terms to the debtor
+    ///     This function will:
+    ///         RES1. Create a claim on BullaClaim with the specified parameters in calldata
+    ///         RES2. Store the loanTerms as indexed by newly created claimId
+    ///         RES3. Emit a FinancingOffered event with the newly created claimId and the terms from calldata
+    ///     Given the following:
+    ///         P1. `msg.value == fee`
+    ///         P2. `msg.sender == claim.creditor`
+    ///         P3. `(terms.minDownPaymentBPS * claim.claimAmount / 10_000) > 0`
+    ///         P4. `terms.termCompletionTimestamp < type(uint40).max`
+    ///         P5. `terms.minDownPaymentBPS < type(uint24).max`
+    ///         P6. `terms.interestBPS < type(uint24).max`
+    ///         P7. `terms.termCompletionTimestamp < type(uint40).max`
+    ///         P8. `terms.termCompletionTimestamp > block.timestamp` TODO: necessary?
+    function createInvoiceWithFinanceOffer(BullaBanker.ClaimParams memory claim, FinanceTerms memory terms)
+        public
+        virtual
+        returns (uint256 claimId);
+
+    /// @param claimId the id of the underlying claim
+    /// @param terms financing terms
+    /// @notice SPEC:
+    ///     Allows a creditor to offer financing on an existing pending claim OR update previously offerred financing terms // TODO: should this function be used to rescind a financing offer?
+    ///     This function will:
+    ///         RES1. Overwrite the `terms` as indexed by the specified `claimId`
+    ///         RES2. Emit a FinancingOffered event
+    ///     Given the following:
+    ///         P1. `claim.status == ClaimStatus.Pending`
+    ///         P2. `msg.sender == claim.creditor`
+    ///         P3. if terms[claimId].termCompletionTimestamp == 0 (implying new terms on an existing claim) ensure msg.value == fee
+    ///         P4. `(terms.minDownPaymentBPS * claim.claimAmount / 10_000) > 0`
+    ///         P5. `terms.minDownPaymentBPS < type(uint24).max`
+    ///         P6. `terms.interestBPS < type(uint24).max`
+    ///         P7. `terms.termCompletionTimestamp < type(uint40).max`
+    ///         P8. `terms.termCompletionTimestamp > block.timestamp` TODO: necessary?
+    function offerFinancing(uint256 claimId, FinanceTerms memory terms) public virtual;
```

Comment:
for PoC, we don't need this. I don't think we have to edit either

Reply:
Sure it is, but we are doing a proof of concept. Is this necessary to prove the concept?

Reply:
We can always add it later.